core/frontend/src/render/webgl/glsl/Clipping.ts

```diff
@@ -66,6 +67,43 @@ const applyClipPlanesPostlude = `
     } else if (!clippedByCurrentPlaneSet && calcClipPlaneDist(v_eyeSpace, plane) < 0.0) {
       clippedByCurrentPlaneSet = true;
     }
+
+    //avoiding near and far clip planes
+    if (i <= u_clipParams[1] - 2) {
+
+      /* 
+      * The closest point on a plane from a point, p,  will be: p minus the distance between p and the plane, in the direction of the plane's normal vector.
+      * We have normal as plane.xyz, and our point as v_eyeSpace.
+      * We can find the distance from the plane using calcClipPlaneDist, then multiply that by the plane's normal.
+      * Subtract the result from the original point to obtain the location of the closest point on the clip plane to v_eyeSpace.
+      * Finally, convert the point on the plane to window coords, and take the distance between that and gl_FragCoord,
+      * to determine whether or not to highlight the current fragment. 
+      */
+
+      float d = calcClipPlaneDist(v_eyeSpace, plane);
+      vec4 pointOnPlane = vec4((v_eyeSpace - (d * plane.xyz)), 1.0);
+
+      //Converting original point to window coords
+      //Only doing this now to ensure the conversion is correct by later comparing with gl_FragCoord
+      //In final implementation, pointOnPlane will be converted, not v_eyeSpace
+
+      vec4 convertedEyeSpace = vec4(v_eyeSpace, 1.0);
+      convertedEyeSpace = convertedEyeSpace * u_proj;
+      convertedEyeSpace.xyz /= convertedEyeSpace.w;                               //Should now be in NDC
```

Comment:
When testing after this point, it's clear that convertedEyeSpace is not in NDC as its coordinate ranges are not -1 to 1. I believe that u_proj is probably at fault, but I have yet to find how.

Reply:
Based on later conversations, disregard my previous comment.